File: 27-remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        left = 0
        right = len(nums) - 1
        while left <= right:
            if nums[right] == val:
                nums.pop()
                right -= 1
            elif nums[left] == val:
                nums[left], nums[right] = nums[right], nums[left]
            else:
                left += 1
                
        return len(nums)

File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int left=0;
        int right=nums.size()-1;
        while(left<=right){
            if(nums[right]==val) right-=1;
            else if(nums[left]!=val) left+=1;
            else swap(nums[left],nums[right]);

        }
        return right+1;
    }
};